### PR TITLE
Added resources for Environment + Environment Secrets

### PR DESCRIPTION
**Pull request by nadenf**
Fixes:

#528 - [feature request] add deployment environment resource
#681 - github_actions_secret: support environment secrets

### CODE REVIEW

File: github/resource_github_repository_environment.go [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+	ctx := context.WithValue(context.Background(), ctxId, d.Id())
+
+	log.Printf("[DEBUG] Reading repository environment: %s (%s/%s/%s)", d.Id(), owner, repoName, envName)
+	env, _, err := client.Repositories.GetEnvironment(ctx, owner, repoName, envName)
+	if err != nil {
+		if ghErr, ok := err.(*github.ErrorResponse); ok {
+			if ghErr.Response.StatusCode == http.StatusNotModified {
+				return nil
+			}
```
Comment (jcudit) [inline github/resource_github_repository_environment.go:118]: nit: do we need this handling?  Surprised that a `3xx` is surfaced as an error.  
👌🏾 to keep but threw us off while reviewing.

File: github/resource_github_actions_environment_secret.go [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+				Optional:  true,
+				ForceNew:  true,
+				Sensitive: true,
+			},
+			"plaintext_value": {
+				Type:      schema.TypeString,
+				Optional:  true,
+				ForceNew:  true,
+				Sensitive: true,
+			},
```
Comment (jcudit) [inline github/resource_github_actions_environment_secret.go:47]: It seems like a [`ConflictsWith`](https://pkg.go.dev/github.com/hashicorp/terraform/helper/schema#Schema.ConflictsWith) field would be helpful here.  Can we get this added to guard against configuring both values?

**Review by jcudit — COMMENTED** [review]
We reviewed this as a group and come up with a couple of suggestions for next steps but this doesn't look too far off from shipping.  The main points we should address are:
- Adding `ConflictsWith`
- Edits to the `vendor/` folder

File: github/resource_github_actions_environment_secret.go [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateSecretNameFunc,
+			},
+			"encrypted_value": {
+				Type:      schema.TypeString,
+				Optional:  true,
+				ForceNew:  true,
+				Sensitive: true,
```
Comment (jcudit) [inline github/resource_github_actions_environment_secret.go:40]: See also https://github.com/integrations/terraform-provider-github/pull/807/files#r645477509 as it applies here.

**Comment by nadenf** [comment]
@jcudit @tibbes  .. All changes made. Hopefully in time for 4.11 ?

**Review by jcudit — APPROVED** [review]
🚀

**Comment by jcudit** [comment]
Ah, just realized we need to update `website.erb` as well.  Feel free to cherry-pick 4e8e9765 and this can go out with the upcoming release.

**Comment by nadenf** [comment]
@jcudit  .. Done !

**Pull request merged**